pycbc/io/live.py

```diff
@@ -332,6 +369,21 @@ def upload(self, fname, gracedb_server=None, testing=True,
             pylab.xlabel('Frequency (Hz)')
             pylab.ylabel('ASD')
             pylab.savefig(psd_series_plot_fname)
+            pylab.close()
+
+        if self.probabilities is not None:
+            prob_fname = fname.replace('.xml.gz', '_probs.json')
+            prob_plot_fname = prob_fname.replace('.json', '.png')
+            probabilities = json.loads(self.probabilities)
+            labels = list(probabilities.keys())
+            sizes = [probabilities[key] for key in probabilities]
```

Comment:
This should be
```python
sizes = [probabilities[key] for key in labels]
```
otherwise you are depending on Python ordering dict keys consistently!

Reply:
Same for the line just after this.

Reply:
I tried to figure out making this a one-liner. I don't think I like the syntax :-)

`labels, sizes = zip(*[(key, probabilities[key]) for key in probabilities])`

Reply:
There used to be something like `probabilities.items()` a while back... Maybe not anymore.

Reply:
Yeah ... I think that's still there (but needs to be cast to a list as python3 will return some weird object).

Reply:
In fact `labels, sizes = zip(*probabilities.items())` seems to work fine on Python 3. Not sure it is a great example of readable code though :)

Reply:
zip(*stuff) is well known Python procedure .. so i'd be OK with that